src/main/scala/edu/berkeley/ce/sparkrocks/Block.scala

```diff
@@ -446,84 +556,26 @@ case class Block(center: Array[Double], faces: Seq[Face], generation: Int=0) ext
   }
 
   /**
-    * Mesh the faces using Delaunay triangulation. This meshing is done
-    * in order to calculate the volume and centroid of the block
+    * Get the counter-clockwise oriented vertices of each face of the block
     *
-    * @return A Seq of seqs that give the local to global mapping for the
-    * triangulation of each of the faces.
+    * @return A mapping from each face of the block to a Seq of vertices for that face
+    *         in a counter-clockwise orientation.
+    *         This function should only be called once all redundant faces have been removed.
     */
-  def meshFaces(vertices: Map[Face, Seq[Array[Double]]]): Map[Face, Seq[Array[Int]]] =
-    // Determine rotation matrix to rotate faces perpendicular to z-axis. This way all vertices
-    // are only in the x-y plane which makes triangulation easier.
-    faces.zip (
-      faces.map { face =>
-        val R = Block.rotationMatrix(face.normalVec, Array(0.0, 0.0, 1.0))
-        val rotatedVertices = vertices(face).map { vertex =>
-          val rotatedVertex = R * DenseVector(vertex)
-          Delaunay.Vector2(rotatedVertex(0), rotatedVertex(1))
-        }.distinct.toList
-        assert(rotatedVertices.nonEmpty)
-        // If normal is -z-axis, order needs to be reversed to maintain clockwise orientation since
-        // rotation matrix is identity matrix in this case - vectors are parallel. Rotation matrix
-        // takes care of all other cases.
-        if (math.abs(face.c + 1.0) < NumericUtils.EPSILON) {
-          Delaunay.Triangulation(rotatedVertices).map { vert =>
-            Array(vert._3, vert._2, vert._1)
-          }
-        } else {
-          Delaunay.Triangulation(rotatedVertices).map { vert =>
-            Array(vert._1, vert._2, vert._3)
-          }
-        }
-      }
-    ).toMap
+  def orientedVertices: Map[Face, Seq[Array[Double]]] = {
```

Comment:
I know we discussed the issue of where to put this logic briefly yesterday.

I think we should make it a method of the `Block` class rather than the `Block` object. And let's call it `orientedVertices`, so a method call would look like `myBlock.orientedVertices`. Does that sound okay to you?